android/engine/src/test/java/org/smartregister/fhircore/engine/task/FhirTaskExpireWorkerTest.kt

```diff
@@ -66,10 +66,15 @@ class FhirTaskExpireWorkerTest : RobolectricTest() {
   @get:Rule(order = 1) val coroutineTestRule = CoroutineTestRule()
   private val fhirEngine: FhirEngine = mockk(relaxed = true)
   private val defaultRepository: DefaultRepository = mockk(relaxed = true)
+  private val fhirResourceClosureUtil: FhirResourceClosureUtil = mockk()
 
   @BindValue
   var fhirTaskUtil: FhirTaskUtil =
-    FhirTaskUtil(ApplicationProvider.getApplicationContext(), defaultRepository)
+    FhirTaskUtil(
+      ApplicationProvider.getApplicationContext(),
+      defaultRepository,
+      fhirResourceClosureUtil,
```

Comment:
Can we include the `resource closure` implementation in the `FhirTaskUtil` instead of  passing utility class as constructor parameter to another utility class. We can rename `FhirTaskUtil` to `FhirResourceUtil` go be more generic.

Reply:
Makes sense